relayer/src/ethereum_checkpoints/tests/mod.rs

```diff
@@ -0,0 +1,440 @@
+    include_bytes!("./sepolia-finality-update-5_254_112.json");
+const FINALITY_UPDATE_5_263_072: &[u8; 4_941] =
+    include_bytes!("./sepolia-finality-update-5_263_072.json");
+
+async fn common_upload_program(
+    client: &GearApi,
+    code: Vec<u8>,
+    payload: impl Encode,
+) -> Result<([u8; 32], [u8; 32])> {
+    let encoded_payload = payload.encode();
+    let gas_limit = client
+        .calculate_upload_gas(None, code.clone(), encoded_payload, 0, true)
+        .await?
+        .min_limit;
+    println!("init gas {gas_limit:?}");
+    let (message_id, program_id, _) = client
+        .upload_program(
+            code,
+            gclient::now_micros().to_le_bytes(),
+            payload,
+            gas_limit,
+            0,
+        )
+        .await?;
+
+    Ok((message_id.into(), program_id.into()))
+}
+
+async fn upload_program(
+    client: &GearApi,
+    listener: &mut EventListener,
+    payload: impl Encode,
+) -> Result<[u8; 32]> {
+    let (message_id, program_id) =
+        common_upload_program(client, WASM_BINARY.to_vec(), payload).await?;
+
+    assert!(listener
+        .message_processed(message_id.into())
+        .await?
+        .succeed());
+
+    Ok(program_id)
+}
+
+#[tokio::test]
+async fn init_and_updating() -> Result<()> {
+    let client_http = Client::new();
+
+    let rpc_url = env::var("RPC_URL").unwrap_or(RPC_URL.into());
+
+    // use the latest finality header as a checkpoint for bootstrapping
+    let finality_update = utils::get_finality_update(&client_http, &rpc_url).await?;
+    let current_period = eth_utils::calculate_period(finality_update.finalized_header.slot);
+    let mut updates = utils::get_updates(&client_http, &rpc_url, current_period, 1).await?;
+
+    println!(
+        "finality_update slot = {}, period = {}",
+        finality_update.finalized_header.slot, current_period
+    );
+
+    let update = match updates.pop() {
+        Some(update) if updates.is_empty() => update.data,
+        _ => unreachable!("Requested single update"),
+    };
+
+    let checkpoint = update.finalized_header.tree_hash_root();
+    let checkpoint_hex = hex::encode(checkpoint);
+
+    println!(
+        "checkpoint slot = {}, hash = {}",
+        update.finalized_header.slot, checkpoint_hex
+    );
+
+    let bootstrap = utils::get_bootstrap(&client_http, &rpc_url, &checkpoint_hex).await?;
+
+    let signature = <G2 as ark_serialize::CanonicalDeserialize>::deserialize_compressed(
+        &update.sync_aggregate.sync_committee_signature.0 .0[..],
+    )
+    .unwrap();
+    let sync_update = utils::sync_update_from_update(signature, update);
+
+    println!("bootstrap slot = {}", bootstrap.header.slot);
+
+    let pub_keys = utils::map_public_keys(&bootstrap.current_sync_committee.pubkeys);
+    let network = match env::var("NETWORK") {
+        Ok(network) if network == "Holesky" => Network::Holesky,
+        Ok(network) if network == "Mainnet" => Network::Mainnet,
+        _ => Network::Sepolia,
```

Comment:
Panic here on unknown network?

Reply:
Also I think it'll be better to define `network` as a `const` because tests will be more isolated of environment variables